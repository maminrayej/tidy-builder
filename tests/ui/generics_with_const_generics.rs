#[derive(tidy_builder::Builder)]
struct MyStruct<'a, 'b: 'a, 'c, const N: usize, const FLG: bool, T: std::fmt::Debug>
where
    T: std::fmt::Display,
    'c: 'a,
{
    req1: &'a T,
    req2: &'b T,
    opt1: Option<&'c T>,
}

fn main() {
    let req1 = "req1".to_string();
    let req2 = "req2".to_string();
    let opt1 = "opt1".to_string();

    let my_struct: MyStruct<0, false, String> = MyStruct::builder()
        .opt1(&opt1)
        .build();

    assert_eq!(my_struct.req1, &req1);
    assert_eq!(my_struct.req2, &req2);
    assert_eq!(my_struct.opt1, Some(&opt1));
}
