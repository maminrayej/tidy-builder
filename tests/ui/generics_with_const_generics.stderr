error[E0599]: no method named `build` found for struct `MyStructBuilder<'_, '_, '_, {_: usize}, {_: bool}, false, false, String>` in the current scope
  --> tests/ui/generics_with_const_generics.rs:19:10
   |
1  | #[derive(tidy_builder::Builder)]
   |          --------------------- method `build` not found for this
...
19 |         .build();
   |          ^^^^^ method not found in `MyStructBuilder<'_, '_, '_, {_: usize}, {_: bool}, false, false, String>`
   |
   = note: the method was found for
           - `MyStructBuilder<'a, 'b, 'c, N, FLG, true, true, T>`
